lib/osf-components/addon/components/contributor-list/component.ts

```diff
@@ -71,6 +81,46 @@ export default class ContributorList extends Component {
         blocker.done();
     });
 
+    @task({ withTestWaiter: true, enqueue: true })
+    removeMeTask = task(function *(this: ContributorList) {
+        if (!this.node || this.node.isAnonymous || !this.currentUser.currentUserId) {
+            return;
+        }
+
+        const userID = this.currentUser.currentUserId;
+        let contributor = this.displayedContributors
+            .find(contrib => contrib.users.get('id') === this.currentUser.currentUserId);
+
+        if (!contributor) {
+            contributor = yield this.store.findRecord('contributor', `${this.node.id}-${userID}`);
+            this.setProperties({
+                displayedContributors: [...this.displayedContributors, contributor],
+                totalContributors: this.totalContributors + 1,
+            });
+        }
+
+        try {
+            yield contributor!.destroyRecord();
+        } catch (e) {
+            const errorMessage = this.intl.t('contributor_list.remove_contributor.error');
+            captureException(e, { errorMessage });
+            this.toast.error(getApiErrorMessage(e), errorMessage);
```

Comment:
You might want to do what we did [here](https://github.com/CenterForOpenScience/ember-osf-web/blob/12da21660708476a8843f092e45a88c0b2bfb575/lib/osf-components/addon/components/contributors/manager/component.ts#L126-L128) and add some context to the toast error.

Reply:
Both achieve the same thing. `this.toast.error(getApiErrorMessage(e), errorMessage);` is more consistent with other `this.toast.error()` usages in the codebase, the first argument is the toast message body; the second is the toast title.